### ISSUE TITLE
Support using native colors for code hosts running Sourcegraph extensions

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#568]
Extensions can now define different colors for decorations on light vs. dark themes. They should also be able to refer to colors from the color palette of the client application they're running on (e.g., GitHub, GitLab, Sourcegraph).

From https://github.com/sourcegraph/sourcegraph-extension-api/pull/110#issuecomment-433773783.

#### Possible impl

For example, you could make a decoration that renders links after each line, and the links would use the exact link color used by GitHub, GitLab, or Sourcegraph:

```typescript
const mydecoration: sourcegraph.TextDocumentDecoration = {
  range: ...,
  after: {
    color: 'var(--link-color)'
  }
}
```

These CSS vars would need to be defined in advance and would be (a logical) part of the Sourcegraph extension API.

**Comment by felixfbecker** [comment]
Already doing this in TypeScript

https://github.com/sourcegraph/sourcegraph-typescript/blob/d6f335c2c6cf3a2f4ecded90595d0770386882aa/extension/src/lsp-conversion.ts#L51-L56

**Comment by felixfbecker** [comment]
I think what's missing here is adding documentation and maybe exposing a string enum in the API to reference these.

As a comparison, VS Code has [`ThemeColor`](https://code.visualstudio.com/api/references/vscode-api#ThemeColor) in the API for this (string only) and a [reference](https://code.visualstudio.com/api/references/theme-color).

**Comment by ryankscott** [comment]
No longer relevant as we intend to sunset Sourcegraph extensions.

**Comment by felixfbecker** [comment]
@ryankscott  Closing as "not planned" instead of as "completed"

**Comment by ryankscott** [comment]
@felixfbecker wow TIL thanks!
